av1an-cli/src/lib.rs

```diff
@@ -442,164 +442,212 @@ fn confirm(prompt: &str) -> io::Result<bool> {
   }
 }
 
-pub fn parse_cli(args: CliOpts) -> anyhow::Result<EncodeArgs> {
-  let temp = if let Some(path) = args.temp.as_ref() {
-    path.to_str().unwrap().to_owned()
-  } else {
-    format!(".{}", hash_path(args.input.as_path()))
-  };
+/// Given Folder and File paths as inputs
+/// Converts them all to file paths
+/// Converting only depth 1 of Folder paths
+pub fn convert_input(args_paths: Vec<PathBuf>) -> Vec<PathBuf> {
```

Comment:
I don't think it's necessary to take a `Vec<PathBuf>` by here, since we're not actually consuming the value (only iterating over it)